libs/wai-utilities/src/Network/Wai/Utilities/Server.hs

```diff
@@ -184,21 +166,39 @@ measureRequests m rtree = withPathTemplate rtree $ \p ->
       requestCounter m p . duration 30 12 m p
 {-# INLINEABLE measureRequests #-}
 
+-- | Alias for @'catchErrors'' True@.
+catchErrors  :: Logger -> OnErrorMetrics -> Middleware
+catchErrors = catchErrors' True
+{-# INLINEABLE catchErrors #-}
+
 -- | Create a middleware that catches exceptions and turns
 -- them into appropriate 'Error' responses, thereby logging
 -- as well as counting server errors (i.e. exceptions that
 -- yield 5xx responses).
-catchErrors :: Logger -> Metrics -> Middleware
-catchErrors l m app req k =
-    app req k `catch` errorResponse
+--
+-- The boolean determines whether response values with
+-- status >= 400 are allowed (servant), or whether we should
```

Comment:
I think I may be missing some context here; why do we want to assert that ALL errors are exceptions and not just a simple 400 response? Doesn't really seem like a big deal to me, but I'm sure I'm missing something.

Reply:
catchErrors is designed to log and analysie exceptions, not responses.  Also, the exceptions have more structure.